src/terms/Hamiltonian.jl

```diff
@@ -229,5 +231,5 @@ end
 function hamiltonian_with_total_potential(Hk::HamiltonianBlock, V)
     operators = [op for op in Hk.operators if !(op isa RealSpaceMultiplication)]
     push!(operators, RealSpaceMultiplication(Hk.basis, Hk.kpoint, V))
-    HamiltonianBlock(Hk.basis, Hk.kpoint, operators, Hk.scratch)
+    HamiltonianBlock(Hk.basis, Hk.kpoint, operators)
```

Comment:
Why remove this? Saves quite a bit of allocation for potential-based SCFs

Reply:
OK, I added it back. It's a bit ugly, but this is a pretty internal thing anyway